dymos/phase/phase.py

```diff
@@ -1126,7 +1126,19 @@ def add_boundary_constraint(self, name, loc, constraint_name=None, units=None,
             raise ValueError(f'Invalid boundary constraint location "{loc}". Must be '
                              '"initial" or "final".')
 
-        if constraint_name is None:
+        expr_operators = ['(', '+', '-', '/', '*', '&', '%', '@']
+        if '=' in name:
+            is_expr = True
+        elif '=' not in name and any(opr in name for opr in expr_operators):
+            raise ValueError(f'The expression provided {name} has invalid format. '
+                             'Expression may be a single variable or an equation'
+                             'of the form "constraint_name = func(vars)"')
+        else:
```

Comment:
For coverage, you should include a test for every error message that you added. They can be simple contrived tests that live in the tests folder in the code rather than something complex in the examples.